### ISSUE TITLE
adaptive learning rate

### ISSUE THREAD

**Issue by ih4cku** [ih4cku/blog#38]
- [Chapter 4. Beyond Gradient Descent](https://www.safaribooksonline.com/library/view/fundamentals-of-deep/9781491925607/ch04.html)
- [Alec Radford's animations for optimization algorithms](http://www.denizyuret.com/2015/03/alec-radfords-animations-for.html)
- [An overview of gradient descent optimization algorithms](http://sebastianruder.com/optimizing-gradient-descent/index.html)
- [Caffe solvers](http://caffe.berkeleyvision.org/tutorial/solver.html)
- [COMPARISON: SGD VS MOMENTUM VS RMSPROP VS MOMENTUM+RMSPROP VS ADAGRAD](http://www.erogol.com/comparison-sgd-vs-momentum-vs-rmsprop-vs-momentumrmsprop/)
- [What are differences between update rules like AdaDelta, RMSProp, AdaGrad and AdaM?](https://www.quora.com/What-are-differences-between-update-rules-like-AdaDelta-RMSProp-AdaGrad-and-AdaM)

**Comment by ih4cku** [comment]
- https://zhuanlan.zhihu.com/p/22252270
- http://climin.readthedocs.io/en/latest/index.html
- https://blog.wtf.sg/2014/08/28/implementing-adadelta/
- http://www.cnblogs.com/neopenx/p/4768388.html
- http://downhill.readthedocs.io/en/stable/index.html
- http://colinraffel.com/wiki/stochastic_optimization_techniques

_BN_
- http://shuokay.com/2016/05/28/batch-norm/
- http://colinraffel.com/wiki/stochastic_optimization_techniques
